examples/peft_llm_gnn/llm_gnn_model.py

```diff
@@ -59,9 +61,34 @@ def get_lm_node_feats(g, lm_feat_names, lm_ntypes):
 
     return lm_feats
 
+class DummyNodeInputLayer(nn.Module):
+    def __init__(self):
+        super(DummyNodeInputLayer, self).__init__()
 
-class LLMGraphModel(gsmodel.GSgnnNodeModelBase):
-    """ LLMGNN Model Class
+    def require_cache_embed(self):
+        return False
+    
+    def forward(self, input_feats, input_nodes):
+        return input_feats
+
+class CosineRandomNegative(nn.Module):
```

Comment:
why does it have cosine in the name?

Reply:
Originally, the normalization is done in the loss function. Now it's in the model computation.

Reply:
I added a assert to ensure this

Reply:
i don't know what you mean here. do you mean you normalize the embeddings? so it's equivalent to compute cosine similarity?

Reply:
yes, cosine similarity is equivalent as dot-product with normalized vectors. And we use LinkPredictDotDecoder here.